uaclient/entitlements/base.py

```diff
@@ -28,6 +28,15 @@
 
 
 class UAEntitlement(metaclass=abc.ABCMeta):
+    @property
+    @abc.abstractmethod
+    def help_doc_url(self) -> str:
+        """Optional URL for top-level product service information
+
+        Empty string if not-applicable.
```

Comment:
Why not `None`? And could this just be set to the not-applicable value, instead of making it abstract and requiring every entitlement to override it?

Reply:
Yeah that was me not thinking clearly on this regarding mypy. tried keeping it the same type instead of the Optional declaration. Fixed.